src/Zip.hs

```diff
@@ -64,26 +70,30 @@ writeZips BucketFiles{..} =
 
 writeZip :: (MonadThrow m, MonadResource m) => Text -> m ()
 writeZip pathToFile = do
-  let stringPath = unpack pathToFile
-  let nameOfZip = (takeFileName . dropTrailingPathSeparator $ stringPath) <> ".zip"
+  let stringPath = dropTrailingPathSeparator . unpack $ pathToFile
+  let nameOfZip = takeFileName stringPath <> ".zip"
   void . register $ deleteFileIfExists nameOfZip
   (liftIO . doesDirectoryExist) stringPath >>= \case
-    True -> liftIO . writeFolderToZip stringPath $ nameOfZip
+    True -> liftIO $
+      getFilesFromFolder stringPath >>= writeFileOrFolderToZip nameOfZip stringPath
     False -> (liftIO . doesFileExist) stringPath >>= \case
-      True -> liftIO . writeFileToZip stringPath $ nameOfZip
+      True -> liftIO $ writeFileOrFolderToZip nameOfZip (takeDirectory stringPath) [stringPath]
       False -> throwM $ FileOrFolderDoesNotExist pathToFile
 
-writeFileToZip :: FilePath -> FilePath -> IO ()
-writeFileToZip path nameOfZip = do
-  archive <- addFilesToArchive [OptRecursive] emptyArchive [path]
-  logEvaporate $ "Zipping " <> pack path
+writeFileOrFolderToZip :: FilePath -> FilePath -> [FilePath] -> IO ()
+writeFileOrFolderToZip nameOfZip rootPath filePaths = do
+  archive <- foldM
+    (\archive path -> do
```

Comment:
I'd pull out this anonymous function into a let or where binding, call it `go` or `folder` if you can't think of a more semantically appropriate name.